oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -133,27 +151,47 @@ public AccessToken refreshAccessToken() throws IOException {
     return new AccessToken(accessToken, new Date(expiresAtMilliseconds));
   }
 
-  /**
-   * Return whether code is running on Google Compute Engine.
-   */
-  static boolean runningOnComputeEngine(HttpTransportFactory transportFactory) {
-    try {
-      GenericUrl tokenUrl = new GenericUrl(METADATA_SERVER_URL);
-      HttpRequest request =
-          transportFactory.create().createRequestFactory().buildGetRequest(tokenUrl);
-      HttpResponse response = request.execute();
-      // Internet providers can return a generic response to all requests, so it is necessary
-      // to check that metadata header is present also.
-      HttpHeaders headers = response.getHeaders();
-      if (OAuth2Utils.headersContainValue(headers, "Metadata-Flavor", "Google")) {
-        return true;
+  /** Return whether code is running on Google Compute Engine. */
+  static boolean runningOnComputeEngine(
+      HttpTransportFactory transportFactory, DefaultCredentialsProvider provider) {
```

Comment:
Do you really want to pass in a DefaultCredentialsProvider to perform env lookup?, 
Would something simpler suffice instead of tying the two classes together.